QueryBuilder/Compilers/SqlServerCompiler.cs

```diff
@@ -60,8 +65,160 @@ public SqlServerCompiler()
             return result;
         }
 
+        private class SqlServerAggregatePercentileApproxColumn : SqlKata.AggregatePercentileApproxColumn
+        {
+            public SqlServerAggregatePercentileApproxColumn() : base() { }
+
+            public SqlServerAggregatePercentileApproxColumn(SqlServerAggregatePercentileApproxColumn other)
+                : base(other)
+            {
+                GroupByClauses = other.GroupByClauses;
+            }
+
+            public override AbstractClause Clone()
+            {
+                return new SqlServerAggregatePercentileApproxColumn(this);
+            }
+
+            public override string Compile(SqlResult ctx)
+            {
+                // percentile_cont(0.9) within group (order by "value") over (partition by ...)
+                var column = new Column { Name = Column }.Compile(ctx);
+                var partition = String.Join(", ", GroupByClauses.Select(clause => clause.Compile(ctx)));
+                partition = partition.Length > 0 ? "PARTITION BY " + partition : "";
+                return $"{Type.ToUpperInvariant()}({Percentile}) WITHIN GROUP(ORDER BY {column}) OVER({partition}) {ctx.Compiler.ColumnAsKeyword}{ctx.Compiler.WrapValue(Alias ?? Type)}";
```

Comment:
`Type' is apparently different here for different implementations.

order by on this aggregate might not work now

Reply:
Fixed by making it a read-only property on the base class.